diff --git a/tests/functional/z_cluster/nodes/test_nodes_restart.py b/tests/functional/z_cluster/nodes/test_nodes_restart.py
index 621bd985d..e6da2228e 100644
--- a/tests/functional/z_cluster/nodes/test_nodes_restart.py
+++ b/tests/functional/z_cluster/nodes/test_nodes_restart.py
@@ -1,7 +1,7 @@
 import logging
 import pytest
-
-from ocs_ci.framework.pytest_customization.marks import brown_squad
+import time
+from ocs_ci.framework.pytest_customization.marks import brown_squad, skipif_ibm_power
 from ocs_ci.framework.testlib import (
     tier4a,
     tier4b,
@@ -93,7 +93,8 @@ class TestNodesRestart(ManageTest):
             wait_for_nodes_status(node_names, constants.STATUS_READY, timeout=420)
         else:
             nodes.restart_nodes_by_stop_and_start(nodes=ocp_nodes, force=force)
-
+        logger.info("Waiting 300 seconds for cluster to stabalize")
+        time.sleep(300)
         self.sanity_helpers.health_check()
         retry(ServiceUnavailable, tries=20, delay=30)(
             self.sanity_helpers.create_resources
